net/tshttpproxy/tshttpproxy.go

```diff
@@ -39,6 +46,59 @@ func setNoProxyUntil(d time.Duration) {
 
 var _ = setNoProxyUntil // quiet staticcheck; Windows uses the above, more might later
 
+// SetSelfProxy configures this package to avoid proxying through any of the
+// provided addresses–e.g. if they refer to proxies being run by this process.
+func SetSelfProxy(addrs ...string) {
```

Comment:
Probably worth a test

Reply:
Added test, and also the integration tests caught a real bug 🎉